lib/tasks/trends.rake

```diff
@@ -29,4 +29,63 @@ namespace :trends do
     end
   end
 
+  desc 'Reduces amount of points for each trend group for the last month'
+  task :reduce => :environment do
+    trend_lines = TrendCounter.order(:trend_id).group_by {|counter| counter.trend_id}
+
+    trend_line_count = 0
+    trend_lines.each do |trend_id, line|
+      to_remove = []
+
+      sorted_counters = line.sort_by {|counter| counter.created_at}
+
+      sorted_counters.drop(1).each_with_index do |counter, idx|
+        to_remove << counter unless TrendImporter.should_create_counter?(sorted_counters[idx], counter.count, counter.created_at)
+      end
+
+      TrendCounter.destroy(to_remove)
+
+      trend_line_count += 1
```

Comment:
I am counting the number of groups (trend lines), and the `#destroy` will return the number of removed data points.